docs/src/advanced/vue-router-url-sync.md

```diff
@@ -8,126 +8,67 @@ navWeight: 3
 editable: true
 githubSource: docs/src/advanced/vue-router-url-sync.md
 ---
-> NOTE: this guide has not yet been updated for v2
+> NOTE: this guide **has** been updated for v2
 
-In this guide you'll learn how to synchronize your InstantSearch experience with the [official Vue-Router](https://github.com/vuejs/vue-router).
+Currently there's three existing ways how to use InstantSearch routing.
 
-There are two things you'll want to setup in order to have the Vue router play nicely with InstantSearch.
+The first option is putting `:routing="true"` on `ais-instant-search`. This will use the default serialising that doesn't lose any information, but might be a bit verbose
 
-1. Extract some parameters from the router and initialize the search store with it. That way, on the initial load of the search experience, it will display results based on the current route.
+The second option is to put an object of configuration. This object can take `stateMapping`, with the functions `stateToRoute` and `routeToState` to serialise differently, but still use the default routing. This allows to rename things to make them easier to read, without touching how the serialising itself happens.
 
-2. Push the new state of the search to the router as the user interacts with the InstantSearch components.
+Finally, you can also change the URL to use full URLs, rather than just the query string. You need to change the `router` key inside the `routing` object. You can import the default (history) router from `import {history} from 'instantsearch.js/es/lib/routers'`, and modify, like in InstantSearch JS.
 
-## Example app
+[![Edit vue-instantsearch-app](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/q8pmz6n7lj?module=%2Fsrc%2FApp.vue)
 
-A fully working example of synchronizing the Vue router with Vue InstantSearch can be found here: https://github.com/algolia/vue-instantsearch-examples/tree/master/examples/vue-router
+All docs for InstantSearch routing configuration are [here](https://community.algolia.com/instantsearch.js/v2/guides/routing.html).
 
-## Initialize InstantSearch with route parameters
+Finally an option is to use Vue Router. All previous examples will _work_ using Vue Router, as long as they don't conflict, as long as you don't try to do specific Vue Router things which are controlled by InstantSearch, since that might cause infinite loops, and InstantSearch provided query strings won't be available to Vue Router, as soon as the routing changes from its initial deserialization.
```

Comment:
Which kind of setup could trigger an infinite loop? Is it already happened?

Reply:
It was actually with the wrong config, I’ll remove it since it’s not really likely